src/template-linter.ts

```diff
@@ -152,11 +156,12 @@ export default class TemplateLinter {
     }
 
     const linterMeta = project.dependenciesMeta.find((dep) => dep.name === 'ember-template-lint');
-    const linterVersion = linterMeta?.version.split('.')[0] || 'unknown';
 
     let sources = [];
 
     try {
+      const linterVersion = linterMeta?.version ? semver.parse(linterMeta.version) : null;
```

Comment:
@NullVoxPopuli I don't think it make sense to compare minor and patch versions

Reply:
wondering will `semver.parse` throw an errors? In worst case for existing approach we getting `unknown` or any string, but with semver we could throw

Reply:
It does potentially throw an error, having looked through the package, although if your meta data's version isn't semantic versioning, I imagine the user has a very odd, possibly bigger issue (low probability of them running into this by accident)

Reply:
I added a comment to this section -- I believe the desire to return on error is correct